lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -4623,13 +4807,14 @@ class AdjointEmitter final : public SILInstructionVisitor<AdjointEmitter> {
   ///   adj[x0] += tuple_extract adj[y], 0
   ///   ...
   void visitTupleInst(TupleInst *ti) {
-    auto av = takeAdjointValue(ti);
+    auto *bb = ti->SILInstruction::getParentBlock();
```

Comment:
Xcode completions aren't always nice.
```suggestion
    auto *bb = ti->getParentBlock();
```

Reply:
This was done to avoid a compilation error, not an Xcode completion. Do you have a better suggestion?
```
/Users/danielzheng/swift-dev/swift/lib/SILOptimizer/Mandatory/Differentiation.cpp:4818:20: error: non-static member 'getParentBlock' found in multiple base-class subobjects of type 'swift::SILNode':
    class swift::TupleInst -> InstructionBaseWithTrailingOperands<enum SILInstructionKind::TupleInst, class swift::TupleInst, class swift::OwnershipForwardingSingleValueInst> -> InstructionBase<(enum swift::SILInstructionKind)84, class swift::OwnershipForwardingSingleValueInst> -> class swift::OwnershipForwardingSingleValueInst -> class swift::SingleValueInstruction -> class swift::SILInstruction -> class swift::SILNode
    class swift::TupleInst -> InstructionBaseWithTrailingOperands<enum SILInstructionKind::TupleInst, class swift::TupleInst, class swift::OwnershipForwardingSingleValueInst> -> InstructionBase<(enum swift::SILInstructionKind)84, class swift::OwnershipForwardingSingleValueInst> -> class swift::OwnershipForwardingSingleValueInst -> class swift::SingleValueInstruction -> class swift::ValueBase -> class swift::SILNode
    auto *bb = ti->getParentBlock();
                   ^
/Users/danielzheng/swift-dev/swift/include/swift/SIL/SILNode.h:429:18: note: member found by ambiguous name lookup
  SILBasicBlock *getParentBlock() const;
                 ^
/Users/danielzheng/swift-dev/swift/lib/SILOptimizer/Mandatory/Differentiation.cpp:4836:30: error: 'TupleType' does not refer to a value
        if (val.getType().is<TupleType>())
                             ^
/Users/danielzheng/swift-dev/swift/include/swift/AST/Module.h:76:9: note: declared here
  class TupleType;
        ^
/Users/danielzheng/swift-dev/swift/lib/SILOptimizer/Mandatory/Differentiation.cpp:4836:41: error: expected expression
        if (val.getType().is<TupleType>())
                                        ^
```

Reply:
There's a method called `getParent()` also. Have you tried that one?

Reply:
Oops, `getParent()` does work. I'll use that, thanks.